Library/Homebrew/api/formula.rb

```diff
@@ -157,6 +157,147 @@ def self.write_names_and_aliases(regenerate: false)
         Homebrew::API.write_names_file!(all_formulae.keys, "formula", regenerate:)
         Homebrew::API.write_aliases_file!(all_aliases, "formula", regenerate:)
       end
+
+      sig { params(hash: T::Hash[String, T.untyped]).returns(T::Hash[String, T.untyped]) }
+      def self.generate_formula_struct_hash(hash)
+        # The returned hash must match FormulaStruct's layout
+        hash = Homebrew::API.merge_variations(hash).dup
```

Comment:
Genuine question rather than critique: why not return a FormulaStruct directly?

Reply:
No reason, this is a good idea! That also helps fix the coupling, which is good